shared/modules/persistence_module.go

```diff
@@ -196,7 +196,7 @@ type PersistenceReadContext interface {
 	GetIntParam(paramName string, height int64) (int, error)
 	GetStringParam(paramName string, height int64) (string, error)
 	GetBytesParam(paramName string, height int64) ([]byte, error)
-	GetParameter(paramName string, height int64) (any, error)
+	GetAllParams() ([][]string, error)
 
 	// Flags
 	GetIntFlag(paramName string, height int64) (int, bool, error)
```

Comment:
So we use a slice of slices instead of a map to maintain the sorted order from the DB query, this allows us to speed up returning the key-value pairs by not having to sort them again (as maps don't maintain order) when we load them into the `GetV1QueryAllChainParams` response.

I don't want to bring back `GetParameter` as an interface function as the function using type parameters is a lot nicer in my opinion Mr. AI

Reply:
Show that AI who's boss